pkg/controller/cluster/cluster.go

```diff
@@ -413,3 +426,76 @@ func (c *ClusterReconciler) validate(cluster *v1alpha1.Cluster) error {
 	}
 	return nil
 }
+
+// lookupServiceCIDR will try to lookup the serviceCIDR of the cluster.
+// It will first look for the 'kube-apiserver' pod, looking for the --service-cluster-ip-range flag.
+// Otherwise
+func (c *ClusterReconciler) lookupServiceCIDR(ctx context.Context) (string, error) {
+	log := ctrl.LoggerFrom(ctx)
+
+	// find the kube-apiserver pod, and look for the '--service-cluster-ip-range' flag
+	matchingLabels := ctrlruntimeclient.MatchingLabels(map[string]string{
+		"component": "kube-apiserver",
+		"tier":      "control-plane",
+	})
+	listOpts := &ctrlruntimeclient.ListOptions{Namespace: "kube-system"}
+	matchingLabels.ApplyToList(listOpts)
+
+	log.Info("looking up serviceCIDR from kube-apiserver pod")
+
+	var podList v1.PodList
+	if err := c.Client.List(ctx, &podList, listOpts); err != nil {
+		if !apierrors.IsNotFound(err) {
+			return "", err
+		}
+	}
+
+	if len(podList.Items) > 0 {
+		apiServerPod := podList.Items[0]
+		apiServerArgs := apiServerPod.Spec.Containers[0].Args
+
+		for _, arg := range apiServerArgs {
+			if strings.HasPrefix(arg, "--service-cluster-ip-range=") {
+				serviceCIDR := strings.TrimPrefix(arg, "--service-cluster-ip-range=")
+				log.Info("found serviceCIDR from kube-apiserver pod: " + serviceCIDR)
+
+				// validate serviceCIDR
+				_, serviceCIDRAddr, err := net.ParseCIDR(serviceCIDR)
+				if err != nil {
+					log.Error(err, "serviceCIDR is not valid")
+					break
+				}
+				return serviceCIDRAddr.String(), nil
+			}
+		}
+	}
+
+	// Try to look for the serviceCIDR creating a failing service.
+	// The error should contain the expected serviceCIDR
+
+	log.Info("looking up serviceCIDR from a failing service creation")
+
+	failingSvc := v1.Service{
+		ObjectMeta: metav1.ObjectMeta{Name: "fail", Namespace: "default"},
+		Spec:       v1.ServiceSpec{ClusterIP: "1.1.1.1"},
+	}
+
+	if err := c.Client.Create(ctx, &failingSvc); err != nil {
+		splittedErrMsg := strings.Split(err.Error(), "The range of valid IPs is ")
+
+		if len(splittedErrMsg) > 1 {
+			serviceCIDR := splittedErrMsg[1]
+			log.Info("found serviceCIDR from failing service creation: " + serviceCIDR)
+
+			// validate serviceCIDR
+			_, serviceCIDRAddr, err := net.ParseCIDR(serviceCIDR)
+			if err != nil {
+				return "", err
+			}
+			return serviceCIDRAddr.String(), nil
+		}
+	}
+
+	log.Info("cannot find serviceCIDR from lookup")
```

Comment:
nit: I think this should be a warning

Reply:
Yes, I need to investigate a bit because it looks like there is no Warn in the logr.Logger interface.. :/

Reply:
Hm, I think I'll leave this here as an info, just because it looks like there is a "philosophical" discussion going here with logr.Logger (https://github.com/go-logr/logr/issues/335#issuecomment-2521775704)

Basically their argument is all around the concepts explained here: https://ntietz.com/blog/the-only-two-log-levels-you-need-are-info-and-error/

It's a bit opinionated but because of the interface we are using here I'll leave it like this, and maybe we can see if it makes sense to do some changes.